src/common/com/intellij/plugins/haxe/util/HaxeElementGenerator.java

```diff
@@ -73,11 +76,27 @@ public static HaxeVarDeclaration createVarDeclaration(Project myProject, String
     return (HaxeVarDeclaration)haxeClass.getVarDeclarations().iterator().next();
   }
 
+  // XXX: Eventually, this ordering should come from the class order in
+  //      preferences... once we have one.
+  private static List<HaxeNamedComponent> sortNamedSubComponents(List<HaxeNamedComponent> unsorted) {
+    // Can't sort a hashed collection, so we must copy it to an orderable type.
+    List<HaxeNamedComponent> sorted = new ArrayList<HaxeNamedComponent>(unsorted);
+    Collections.sort(sorted, new Comparator<HaxeNamedComponent>() {
+      @Override
+      public int compare(HaxeNamedComponent o1, HaxeNamedComponent o2) {
+        String name1 = o1.getName();
+        String name2 = o2.getName();
+        return name1.compareTo(name2);
+      }
+    });
+    return sorted;
+  }
+
   public static List<HaxeNamedComponent> createNamedSubComponentsFromText(Project myProject, String text) {
     final PsiFile dummyFile = createDummyFile(myProject, HaxeCodeGenerateUtil.wrapFunction(text).getFirst());
     final HaxeClass haxeClass = PsiTreeUtil.getChildOfType(dummyFile, HaxeClass.class);
     assert haxeClass != null;
-    return HaxeResolveUtil.findNamedSubComponents(haxeClass);
+    return sortNamedSubComponents(HaxeResolveUtil.findNamedSubComponents(haxeClass));
```

Comment:
We have to sort them /somehow/ to get the same ordering every time.  Before sorting, they are in a hash table.  That's definitely in an order when we retrieve them, but the order changes between versions of java, causing the ordering to be different when we iterate over the elements (to add them to the class).  A sorted list gives us an ordering that we control, because we control the compare operator.  Sorting the elements to insert by name is a likely thing that engineers would do, so I grabbed at it.  In this case, it's really just putting getXXX before setXXX.  If there are multiple getters and setters being created (like this test does), the get and set for each variable will be together; so getFoo() is followed by setFoo()  and getThis() is followed by setThis().  Before this change whether get or set would come first was essentially random, but really based upon the has algorithm.